sources/packages/web/src/services/http/InstitutionUserApi.ts

```diff
@@ -0,0 +1,151 @@
+import HttpBaseClient from "./common/HttpBaseClient";
+import { PaginatedResults, ClientIdType, PaginationOptions } from "@/types";
+import {
+  InstitutionUserAPIOutDTO,
+  InstitutionUserDetailAPIOutDTO,
+  InstitutionUserStatusAPIOutDTO,
+  CreateInstitutionUserAPIInDTO,
+  UpdateInstitutionUserAPIInDTO,
+  UserActiveStatusAPIInDTO,
+} from "@/services/http/dto";
+import { AuthService } from "../AuthService";
+import { getPaginationQueryString } from "@/helpers";
+
+export class InstitutionUserApi extends HttpBaseClient {
+  /**
+   * Get details of user who is logged in.
+   * @returns user details.
+   */
+  async getMyInstitutionDetails(
+    header?: any,
+  ): Promise<InstitutionUserDetailAPIOutDTO> {
+    return this.getCallTyped<InstitutionUserDetailAPIOutDTO>(
+      this.addClientRoot("institution-user/my-details"),
+      header,
+    );
+  }
+
+  /**
+   * Get institution user summary.
+   * @param paginationOptions pagination options.
+   * @param institutionId institution to be searched. If not
+   * provided the institution id will be retrieved from the token,
+   * when available.
+   * @returns all filtered institution users.
+   */
+  async searchUsers(
+    paginationOptions: PaginationOptions,
+    institutionId?: number,
+  ): Promise<PaginatedResults<InstitutionUserAPIOutDTO>> {
+    let url = "institution-user?";
+    if (AuthService.shared.authClientType === ClientIdType.AEST) {
+      url += `institutionId=${institutionId}&`;
+    }
+    url += getPaginationQueryString(paginationOptions);
+    return this.getCallTyped<PaginatedResults<InstitutionUserAPIOutDTO>>(
+      this.addClientRoot(url),
+    );
+  }
+
+  /**
+   * Create a user, associate with the institution, and assign the authorizations.
+   * @param payload authorizations to be associated with the user.
+   * @param institutionId institution to have the user associated. If not provided the
+   * token information will be used, if available.
+   * @returns Primary identifier of the created resource.
```

Comment:
there is no return right?

Reply:
There is but it is not used. I will remove it.

Reply:
Changed!